src/Avalonia.Controls/TabControl.cs

```diff
@@ -211,15 +211,17 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
         {
             base.OnPointerPressed(e);
 
-            if (e.GetCurrentPoint(this).Properties.IsLeftButtonPressed && e.Pointer.Type == PointerType.Mouse)
+            if (e.GetCurrentPoint(this).Properties.IsLeftButtonPressed && 
+                (e.Pointer.Type == PointerType.Mouse || e.Pointer.Type == PointerType.Pen))
             {
                 e.Handled = UpdateSelectionFromEventSource(e.Source);
             }
         }
 
         protected override void OnPointerReleased(PointerReleasedEventArgs e)
         {
-            if (e.InitialPressMouseButton == MouseButton.Left && e.Pointer.Type != PointerType.Mouse)
+            if (e.InitialPressMouseButton == MouseButton.Left &&
+                (e.Pointer.Type == PointerType.Mouse || e.Pointer.Type == PointerType.Pen))
```

Comment:
It's wrong here. 
Previously we had following log:
1. Mouse pressed was handled by OnPointerPressed to select item. Scrolling is done independently with Wheel.
2. Touch pressed was handled by OnPointerReleased to select item. "Pressed" event was handled by scrolling.
To moment of tab item selection actually depends on pointer type.
With your change both events now handled in Pen and Mouse mode. And no event is handled is Touch mode.

Which behavior is better for Pen? What other framework do?

Reply:
oops. missed !=. :(